File: Java/Persona.java
package Java;

class Persona {
    private final String nombre;
    private boolean infectada;
    private boolean manosContaminadas;

    public Persona(String nombre, boolean infectada) {
        if (nombre == null || nombre.isBlank()) {
            throw new IllegalArgumentException("El nombre no puede estar vacio.");
        }

        this.nombre = nombre;
        this.infectada = infectada;
        this.manosContaminadas = false;
    }

    public String getNombre() {
        return nombre;
    }

    public boolean estaInfectada() {
        return infectada;
    }

    public boolean tieneManosContaminadas() {
        return manosContaminadas;
    }

    void contaminarManos() {
        manosContaminadas = true;
    }

    void lavarManos() {
        manosContaminadas = false;
    }

    void infectar() {
        infectada = true;
    }

    public void mostrarEstado() {
        System.out.printf(
                "Persona: %-8s | Infectada: %-5s | Manos contaminadas: %-5s%n",
                nombre,
                convertirSiNo(infectada),
                convertirSiNo(manosContaminadas));
    }

    private String convertirSiNo(boolean valor) {
        return valor ? "SI" : "NO";
    }
}

File: CPP/Persona.cpp
#include "Persona.h"
#include <iostream>
using namespace std;

Persona::Persona(string nombre, bool infectada) {
    this->nombre = nombre;
    this->infectada = infectada;
    this->manosContaminadas = false;
}

string Persona::getNombre() {
    return nombre;
}

bool Persona::estaInfectada() {
    return infectada;
}

bool Persona::tieneManosContaminadas() {
    return manosContaminadas;
}

void Persona::contaminarManos() {
    manosContaminadas = true;
}

void Persona::lavarManos() {
    manosContaminadas = false;
}

void Persona::infectar() {
    infectada = true;
}

void Persona::mostrarEstado() {
    cout << "Persona: " << nombre
         << " | Infectada: " << (infectada ? "SI" : "NO")
         << " | Manos contaminadas: "
         << (manosContaminadas ? "SI" : "NO")
         << endl;
}